pkg/data/components/RedactionProcessor.yaml

```diff
@@ -0,0 +1,65 @@
+kind: RedactionProcessor
+name: Redaction Processor
+style: processor
+type: base
+status: development
+version: v0.0.1
+summary: A processor that can be used to redact attribute values in telemetry.
+description: |
+  This processor allows you to redact attribute values from traces, metrics, and logs based on
+  specified rules.
+tags:
+  - category:processor
+  - service:collector
+  - category:filter
+  - signal:OTelTraces
+  - signal:OTelMetrics
+  - signal:OTelLogs
+ports:
+  - name: Traces
+    direction: input
+    type: OTelTraces
+  - name: Metrics
+    direction: input
+    type: OTelMetrics
+  - name: Logs
+    direction: input
+    type: OTelLogs
+properties:
```

Comment:
I think we should the `KeysToNotRedact` (aka `ignored_keys`) perhaps as an advanced property.